spec/services/generate_test_data_spec.rb

```diff
@@ -37,5 +37,12 @@
       Faker::Config.random = Random.new(42)
       expect { GenerateTestData.new(1, provider).generate }.to change { ApplicationForm.count }.by(1)
```

Comment:
This is not specific to the audit trail support UI work, but it is not clear to me what this scenario is testing. After our chat, my understading is the fake data generator changed to provide email addresses which are a bit more readable, leading to the errors we can easily fix with Faker::Name.unique. However, the purpose of the spec is to test the code preventing duplicate course codes and creating alternatives. Should the test for this be a lower-level/unit spec, rather than a service/system one?